lib/common/xml.c

```diff
@@ -1536,39 +1551,40 @@ log_xml_node_and_children(GString *buffer, int log_level, const char *prefix,
         return;
     }
 
-    name = crm_element_name(data);
-
-    g_string_truncate(buffer, 0);
-
     if (pcmk_is_set(options, xml_log_option_open)) {
+        g_string_truncate(buffer, 0);
         insert_prefix(options, buffer, depth);
 
         if (data->type == XML_COMMENT_NODE) {
             pcmk__g_strcat(buffer, "<!--", (const char *) data->content, "-->",
                            NULL);
 
-        } else {
-            pcmk__g_strcat(buffer, "<", name, NULL);
+        } else {    // data->type == XML_ELEMENT_NODE
```

Comment:
We already return early if it's not a comment or element. The GitHub diff hides it because it's too far away -- right after the `CRM_ASSERT(buffer != NULL)`.

Thought about changing to a switch, but we already have an `if X return early` block that these fit into, and there are only two.

Reply:
I don't mind changing to switch though

Reply:
It's a switch in the latest push. Maybe not in this exact commit.